metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -619,108 +620,114 @@ public void testMissingInMemorySnapshot() throws Exception {
-                    newController
-                        .replicationControl()
-                        .brokersToIsrs()
-                        .partitionsWithBrokerInIsr(2)
-                ).size();
-                int partitionsWithReplica0 = Utils.toList(
-                    newController
-                        .replicationControl()
-                        .brokersToIsrs()
-                        .partitionsWithBrokerInIsr(0)
-                ).size();
-
-                assertEquals(numPartitions, partitionsWithReplica0);
-                assertNotEquals(0, partitionsWithReplica2);
-                assertTrue(
-                    partitionsWithReplica0 > partitionsWithReplica2,
-                    String.format(
-                        "partitionsWithReplica0 = %s, partitionsWithReplica2 = %s",
-                        partitionsWithReplica0,
-                        partitionsWithReplica2
                     )
-                );
-            }
+            ).get().topics().find(topicName).topicId();
+
+            // Create a lot of alter isr
+            List<AlterIsrRequestData.PartitionData> alterIsrs = IntStream
+                .range(0, numPartitions)
+                .mapToObj(partitionIndex -> {
+                    PartitionRegistration partitionRegistration = controller.replicationControl().getPartition(
+                        topicId,
+                        partitionIndex
+                    );
+
+                    return new AlterIsrRequestData.PartitionData()
+                        .setPartitionIndex(partitionIndex)
+                        .setLeaderEpoch(partitionRegistration.leaderEpoch)
+                        .setCurrentIsrVersion(partitionRegistration.partitionEpoch)
+                        .setNewIsr(Arrays.asList(0, 1));
+                })
+                .collect(Collectors.toList());
+
+            AlterIsrRequestData.TopicData topicData = new AlterIsrRequestData.TopicData()
+                .setName(topicName);
+            topicData.partitions().addAll(alterIsrs);
+
+            int leaderId = 0;
+            AlterIsrRequestData alterIsrRequest = new AlterIsrRequestData()
+                .setBrokerId(leaderId)
+                .setBrokerEpoch(brokerEpochs.get(leaderId));
+            alterIsrRequest.topics().add(topicData);
+
+            logEnv.logManagers().get(0).resignAfterNonAtomicCommit();
+
+            int oldClaimEpoch = controller.curClaimEpoch();
+            assertThrows(
+                ExecutionException.class,
+                () -> controller.alterIsr(alterIsrRequest).get()
+            );
+
+            // Wait for the controller to become active again
+            assertSame(controller, controlEnv.activeController());
+            assertTrue(
+                oldClaimEpoch < controller.curClaimEpoch(),
+                String.format("oldClaimEpoch = %s, newClaimEpoch = %s", oldClaimEpoch, controller.curClaimEpoch())
+            );
```

Comment:
Only this should have changed. The rest are indentation changes from the previous commit.